### PR TITLE
velero: Add e2e tests

### PR DESCRIPTION
**Pull request by surajssd**
Install velero on Packet CI pipeline.

### CODE REVIEW

File: test/components/velero/velero_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// Copyright 2021 The Lokomotive Authors
```
Comment (surajssd) [inline test/components/velero/velero_test.go:1]: Make a note 2021, getting used to can take some time.

File: test/components/velero/velero_test.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build packet
+// +build e2e
+
+package rook_test
```
Comment (invidian) [inline test/components/velero/velero_test.go:18]: This shouldn't be `rook_test`.

File: ci/packet/packet-cluster.lokocfg.envsubst [hunk 3]
```diff
@@ -226,3 +226,26 @@ component "web-ui" {
...
+  restic {
+    credentials = <<EOF
+[default]
+aws_access_key_id = "$AWS_ACCESS_KEY_ID"
+aws_secret_access_key = "$AWS_SECRET_ACCESS_KEY"
+EOF
+
+    backup_storage_location {
+      provider = "aws"
+      bucket   = "velero-test-ci"
```
Comment (invidian) [inline ci/packet/packet-cluster.lokocfg.envsubst:242]: This bucket must exist for tests to converge? If so, I think this dependency is quite hidden and we should make it somehow more explicit or document the requirement.

**Review by invidian — CHANGES_REQUESTED** [review]

File: test/components/velero/velero_test.go (hunk 2 above)
Comment (surajssd) [reply test/components/velero/velero_test.go:18]: oops, copy paste error :facepalm:

File: ci/packet/packet-cluster.lokocfg.envsubst (hunk 3 above)
Comment (surajssd) [reply ci/packet/packet-cluster.lokocfg.envsubst:242]: Added a comment on top of this parameter.

File: test/components/velero/velero_test.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+func TestResticDaemonset(t *testing.T) {
+	client := testutil.CreateKubeClient(t)
+
+	t.Run("daemonset", func(t *testing.T) {
+		t.Parallel()
+		daemonset := "restic"
+
+		testutil.WaitForDaemonSet(t, client, namespace, daemonset, testutil.RetryInterval, testutil.Timeout)
+	})
+}
```
Comment (invidian) [inline test/components/velero/velero_test.go:48]: Why 2 separate test functions and then `t.Run` inside each? That doesn't make any sense.

**Review by invidian — CHANGES_REQUESTED** [review]

File: test/components/velero/velero_test.go (hunk 4 above)
Comment (surajssd) [reply test/components/velero/velero_test.go:48]: One is for restic and other is for deployment velero. What's not making sense?

Comment (invidian) [reply test/components/velero/velero_test.go:48]: Using `t.Run()` only once. we should either merge 2 tests or remote `t.Run()`.

**Review by invidian — APPROVED** [review]

**Review by knrt10 — APPROVED** [review]
+1

**Pull request merged**